lib/transcoder/model/video.ex

```diff
@@ -1,11 +1,48 @@
 defmodule Transcoder.Model.Video do
+  @moduledoc """
+  Struct representing video file.
+  """
+
   use TranscoderWeb, :verified_routes
+  alias __MODULE__
+
+  defstruct [:upload, :extension, resolution: :original]
 
-  defstruct [:filename, resolution: :original]
+  @doc """
+  Return new Video for upload path and original filename. Video extension is extracted from filename, and resolution will be :original.
+  """
+  def for_upload(path, filename) do
+    upload_name = Path.basename(path)
+    %Video{upload: upload_name, extension: extension(filename)}
+  end
+
+  @doc """
+  Return copy of given video with resolution set to provided value.
+  """
+  def with_resolution(video, resolution) do
+    %Video{upload: video.upload, extension: video.extension, resolution: resolution}
+  end
 
+  @doc """
+  Return copy of given video with resolution set to :original.
+  """
+  def original(video) do
+    %Video{upload: video.upload, extension: video.extension, resolution: :original}
+  end
```

Comment:
Nit: resolution could be an atom argument

Reply:
wdym? that these should be the same multi-clause function with argument `resolution` that has default value of `:original`?

Reply:
Exactly